pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -78,6 +86,41 @@ public class PipelineManager implements ApplicationListener<ApplicationReadyEven
   // TODO expose this in the web-UI
   private LastRunStatus lastRunStatus = LastRunStatus.NOT_RUN;
 
+  /**
+   * This method publishes the beam pipeline metrics to the spring boot actuator. Previous metrics
+   * are removed from the actuator before publishing the latest metrics
+   *
+   * @param metricResults
+   */
+  void publishPipelineMetrics(MetricResults metricResults) {
+    removePipelineMetrics();
+    MetricQueryResults metricQueryResults = EtlUtils.getMetrics(metricResults);
+    for (MetricResult<Long> metricCounterResult : metricQueryResults.getCounters()) {
+      Counter counter =
+          meterRegistry.counter(
+              metricCounterResult.getName().getNamespace()
+                  + "_"
+                  + metricCounterResult.getName().getName());
+      counter.increment(metricCounterResult.getAttempted());
```

Comment:
Do we need to use `Counter` here (I mean in the context of  `MeterRegistry`)? It seems to me there should be a concept that just reports the current value. With a quick look at the [javadoc](https://www.javadoc.io/static/io.micrometer/micrometer-core/1.11.0/io/micrometer/core/instrument/MeterRegistry.html) I am guessing that a `gauge` maybe is a better concept?

Reply:
As discussed, replacing the `Counter` with `Gauge` as it represents the final value of the metric in the context of actuator. Changes have been made.